tests/test_logging.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import logging
+import re
+from io import StringIO
+from unittest import TestCase
+
+
+import couchdbkit.logging as mod
+from couchdbkit import Server
+
+
+class TestLogging(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestLogging, cls).setUpClass()
+        cls.server = Server()
+        cls.db = cls.server.create_db('couchdbkit_test')
+        doc = {"name": "Yesterday's Heroes"}
+        cls.db.save_doc(doc)
+        cls.doc_id = doc["_id"]
+        cls.log = CaptureLogOutput(mod.request_logger.name)
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestLogging, cls).tearDownClass()
+        cls.log.close()
+        try:
+            del cls.server['couchdbkit_test']
+        except Exception:
+            pass
+
+    def setUp(self):
+        self.log.clear()
+
+    def assertRegex(self, text, regex):
+        assert re.search(regex, text), "%r not matched by %r" % (text, regex)
+
+    def test_no_logging_when_not_installed(self):
+        self.assertFalse(self.log)
+        self.db.get(self.doc_id)
+        self.assertFalse(self.log)
+
+    def test_install_request_logger(self):
+        self.addCleanup(mod.install_request_logger())
+        self.db.get(self.doc_id)
+        self.assertRegex(str(self.log),
+            r"^GET to couchdbkit_test/{} took \d.\d".format(self.doc_id))
```

Comment:
small thing, but installing [freezegun](https://github.com/spulec/freezegun) could make these tests simpler and a bit more complete

Reply:
I'm not comparing dates or times here. Can you elaborate on how that's relevant?

Reply:
isn't the part of the log that requires you to use a regex (because you aren't comparing it) the datetime? i may have misunderstood something here.

Reply:
no, it's the request duration (a number)

Reply:
yeah, a number calculated from `datetime`s (or an equivalent) that can be pinned using freezegun